lib/msprime.c

```diff
@@ -6355,22 +6355,49 @@ msp_dirac_common_ancestor_event(msp_t *self, population_id_t pop_id, label_id_t
  **************************************************************/
 
 static double
-beta_compute_timescale(msp_t *self, population_t *pop)
+beta_compute_juvenile_mean(msp_t *self)
 {
     double alpha = self->model.params.beta_coalescent.alpha;
+    double m;
+    if (self->ploidy > 1) {
+        m = 2 + exp(alpha * log(2) + (1 - alpha) * log(3) - log(alpha - 1));
+    } else {
+        m = 1 + exp((1 - alpha) * log(2) - log(alpha - 1));
+    }
+    return m;
+}
+
+static double
+beta_compute_truncation(msp_t *self)
+{
     double truncation_point = self->model.params.beta_coalescent.truncation_point;
-    double m = 1 + exp(alpha * log(2) + (1 - alpha) * log(3) - log(alpha - 1));
+    double m;
+    if (truncation_point < DBL_MAX) {
```

Comment:
If we set the input value to ``DBL_MAX`` then I think this condition isn't necessary. I think (but we'd need to verify, just t to be sure) that ``BDL_MAX / (DBL_MAX  + m)`` is 1 for any reasonable value of m.

The way to specify DBL_MAX in python is via ``sys.float_info.max``.

Reply:
Is there much harm in keeping it as is? It costs a few extra lines of code and an if-statement, but the value of `m` diverges as alpha approaches 1, so in principle `DBL_MAX / (DBL_MAX + m) == 1` should fail for sufficiently small but otherwise valid values of alpha.

Reply:
Yeah, OK, it's probably safer to have the special case and I guess it all works either way.